client/my-sites/plugins/plugin-install-button/index.jsx

```diff
@@ -257,7 +272,9 @@ PluginInstallButton.propTypes = {
 };
 
 export default connect(
-	null,
+	( state, ownProps ) => ( {
+		siteIsConnected: ownProps.selectedSite && getSiteConnectionStatus( state, ownProps.selectedSite.ID ),
```

Comment:
Optional but it looks like we reuse the id in a few places:
```jsx
const mapStateToProps = ( state, { selectedSite } ) => {
   const siteId = selectedSite && selectedSite.ID;
   return {
       siteId,
       siteIsConnected: getSiteConnectionStatus( state, siteId )
   }
}
```

Likewise in render we can render the query component without a prop check, though this requires adding a check in `QuerySiteConnectionStatus` to do a no-op when siteId is falsey.
```
<QuerySiteConnectionStatus siteId={ siteId } />
```